Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs

```diff
@@ -14,93 +17,161 @@ public class GvrSceneManager : SingletonBehaviour<GvrSceneManager>
+            // i.e. created GOs will be automatically put to right scene afterwards.
+            await Task.Yield();
+            
+            // Remove previous scene if it exists
             if (GameData.I.WorldScene.HasValue)
-            {
                 SceneManager.UnloadSceneAsync(GameData.I.WorldScene.Value);
-            }
 
             GameData.I.WorldScene = newWorldScene;
+            return newWorldScene;
+        }
+        
+        /// <summary>
+        /// Create loading scene and wait for a few milliseconds to go on, ensuring loading bar is selectable.
+        /// Async: execute in sync, but whole process can be paused for x amount of frames.
+        /// </summary>
+        private async Task ShowLoadingScene(string worldName = null)
+        {
+            TextureManager.I.LoadLoadingDefaultTextures();
 
-            WorldCreator.I.Create(newWorldName);
+            generalScene = SceneManager.GetSceneByName(generalSceneName);
+            if (generalScene.isLoaded)
+            {
+                SceneManager.MoveGameObjectToScene(interactionManager, SceneManager.GetSceneByName("Bootstrap"));
+                SceneManager.UnloadSceneAsync(generalScene);
+                generalSceneLoaded = false;
+            }
 
-            SceneManager.sceneLoaded += WorldSceneLoaded;
-            SceneManager.activeSceneChanged += ActiveSceneChanged;
+            SetLoadingTextureForWorld(worldName);
+
+            SceneManager.LoadScene("Loading", new LoadSceneParameters(LoadSceneMode.Additive));
+
+            // Delay for magic number amount to make sure that bar can be found
+            // 1 and 2 caused issues for the 3rd time showing the loading scene in editor
+            await Task.Delay(ensureLoadingBarDelayMilliseconds);
         }
 
-        /// <summary>
-        /// We need to set world scene loaded as active scene. This is the only way OcclusionCulling data is fetched
-        /// for the world in a multi-scene scenario. Also set the player to we have a reference regardless of scene
-        /// </summary>
-        private void WorldSceneLoaded(Scene scene, LoadSceneMode mode)
+        private void SetLoadingTextureForWorld(string worldName)
+        {
+            if (worldName == null)
+                return;
+
+            // set the loading background texture properly
+            // TODO: for new game we need to load texture "LOADING.TGA"
+            var textureString = "LOADING_" + worldName.Split('.')[0].ToUpper() + ".TGA";
+            TextureManager.I.SetTexture(textureString, TextureManager.I.GothicLoadingMenuMaterial);
+        }
+
+        private void HideLoadingScene()
         {
+            SceneManager.UnloadSceneAsync("Loading");
+
+            LoadingManager.I.ResetProgress();
+        }
+        
+        private void OnWorldSceneLoaded(Scene scene, LoadSceneMode mode)
+        {
+            if (scene.name == newWorldName)
```

Comment:
our scene is lower case but newWorldName might be all uppercase
```suggestion
            if (scene.name.toLower() == newWorldName.toLower())
```